File: src/misc/ipf-stats.ts
/*
Sources
https://institutpf.org/assets/Documents/Normes-projection/InstitutPF-NHP-feuillet_fr.pdf

Revised
2026-04-23
*/

export interface ReturnRates {
    SHORT_TERM: number;
    FIXED_INCOME: number;
    CANADIAN_EQUITIES: number;
    US_EQUITIES: number;
    /** @deprecated Use FOREIGN_DEVELOPED_MARKET_EQUITIES instead. Will be removed in a future version. */
    INTL_DEVELOPED_MARKET_EQUITIES: number;
    FOREIGN_DEVELOPED_MARKET_EQUITIES: number;
    EMERGING_MARKET_EQUITIES: number;
    CONSERVATIVE_PORTFOLIO: number;
    BALANCED_PORTFOLIO: number;
    DYNAMIC_PORTFOLIO: number;
}

export interface IPFStatistics {
    INFLATION: number;
    PERFORMANCE_RATE: number;
    YMPE_GROWTH_RATE: number;
    RETURN_RATES: ReturnRates;
    BORROWING_RATE: number;
}

export const IPF: IPFStatistics = {
    INFLATION: 0.021,
    PERFORMANCE_RATE: 0.01,
    YMPE_GROWTH_RATE: 0.031,
    RETURN_RATES: {
        SHORT_TERM: 0.024,
        FIXED_INCOME: 0.032,
        CANADIAN_EQUITIES: 0.063,
        US_EQUITIES: 0.064,
        INTL_DEVELOPED_MARKET_EQUITIES: 0.065,
        FOREIGN_DEVELOPED_MARKET_EQUITIES: 0.065,
        EMERGING_MARKET_EQUITIES: 0.075,
        CONSERVATIVE_PORTFOLIO: 0.027,
        BALANCED_PORTFOLIO: 0.034,
        DYNAMIC_PORTFOLIO: 0.044,
    },
    BORROWING_RATE: 0.044,
};

File: src/commonMain/kotlin/com/equisoft/taxca/misc/IpfStats.kt
/*
Sources
https://institutpf.org/assets/Documents/Normes-projection/InstitutPF-NHP-feuillet_fr.pdf

Revised
2026-04-23
*/

package com.equisoft.taxca.misc

data class ReturnRates(
    val shortTerm: Double,
    val fixedIncome: Double,
    val canadianEquities: Double,
    val usEquities: Double,
    val foreignDevelopedMarketEquities: Double,
    val emergingMarketEquities: Double,
    val conservativePortfolio: Double,
    val balancedPortfolio: Double,
    val dynamicPortfolio: Double,
)

data class IPFStatistics(
    val inflation: Double,
    val performanceRate: Double,
    val ympeGrowthRate: Double,
    val returnRates: ReturnRates,
    val borrowingRate: Double,
)

val Ipf: IPFStatistics = IPFStatistics(
    inflation = 0.021,
    performanceRate = 0.01,
    ympeGrowthRate = 0.031,
    returnRates = ReturnRates(
        shortTerm = 0.024,
        fixedIncome = 0.032,
        canadianEquities = 0.063,
        usEquities = 0.064,
        foreignDevelopedMarketEquities = 0.066,
        emergingMarketEquities = 0.075,
        conservativePortfolio = 0.027,
        balancedPortfolio = 0.034,
        dynamicPortfolio = 0.044,
    ),
    borrowingRate = 0.044,
)
